pkg/virt-handler/vm.go

```diff
@@ -1550,16 +1550,17 @@ func (d *VirtualMachineController) defaultExecute(key string,
 	// set true to ensure that no updates to the current VirtualMachineInstance state will occur
 	forceIgnoreSync := false
 
-	log.Log.Infof("Processing event %v", key)
-	if vmiExists {
-		log.Log.Object(vmi).Infof("VMI is in phase: %v\n", vmi.Status.Phase)
-	} else {
-		log.Log.Info("VMI does not exist")
-	}
-	if domainExists {
-		log.Log.Object(domain).Infof("Domain status: %v, reason: %v\n", domain.Status.Status, domain.Status.Reason)
+	log.Log.V(3).Infof("Processing event %v", key)
+
+	if vmiExists && domainExists {
+		log.Log.Object(vmi).Infof("VMI is in phase: %v | Domain status: %v, reason: %v", vmi.Status.Phase, domain.Status.Status, domain.Status.Reason)
+	} else if vmiExists {
+		log.Log.Object(vmi).Infof("VMI is in phase: %v | Domain does not exist", vmi.Status.Phase)
+	} else if domainExists {
+		vmiRef := v1.NewVMIReferenceWithUUID(domain.ObjectMeta.Namespace, domain.ObjectMeta.Name, domain.Spec.Metadata.KubeVirt.UID)
+		log.Log.Object(vmiRef).Infof("VMI does not exist | Domain status: %v, reason: %v", domain.Status.Status, domain.Status.Reason)
 	} else {
-		log.Log.Info("Domain does not exist")
+		log.Log.Info("VMI does not exist | Domain does not exist")
```

Comment:
logging fidelity is lost here. It's usefull to know if the VMI or Domain exists independently of one another.

Reply:
Do you have any suggestion on how we can approach this?

Reply:
You could do something like
```
log.Log.Infof("VMI does not exist: %v | Domain does not exist: %v", !vmiExists, !domainExists)
```
Or write it the other way round. Just an idea.

Reply:
Hm, I'm not sure if I understand the problem. We handle all cases in separate `else if` statements. So we catch all cases:
1. both domain and vmi exist -> `if`
2. just vmi exists -> first `else if`
3. just domain exists -> second `else if`
4. both domain and vmi do not exist -> `else`

So I don't see a point why we need:
```
log.Log.Infof("VMI does not exist: %v | Domain does not exist: %v", !vmiExists, !domainExists)
```
if we already know that they do not exist in this `else` block.

Reply:
Ah, right. Thanks for mentioning it again, on a second read it makes more sense.
I think the reason this was confusing is, that the two cases are connected by a `|`.
This case means "neither of the two are there" because every other option is already handled above.
Then I only problem I could see is that the independent log messages for domain and vmi are of value?